### ISSUE TITLE
Improve the ReactorBuilder.requiredInterfaces skip message

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5217]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5217|
|Type|enhancement|
|Created|2011-08-03 11:52:03Z|
|Branch|https://github.com/twisted/twisted/tree/required-ifaces-skip-msg-5217|

`twisted.internet.test.reactormixins.ReactorBuilder` supports skipping tests on a particular reactor if it does not provide a required interface.  It does so with a skip message like:

```
<twisted.internet.selectreactor.SelectReactor object at 0x105202ad0> does not provide <InterfaceClass twisted.internet.interfaces.IReactorWin32Events>
```

A better message is probably

_twisted.internet.selectreactor.SelectReactor does not provide twisted.internet.interfaces.IReactorWin32Events_

This is also a win in the test suite output, since trial will combine skips that have the same message.

<details><summary>Searchable metadata</summary>

```
trac-id__5217 5217
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_required_ifaces_skip_msg_5217 branches/required-ifaces-skip-msg-5217
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy_tests easy tests
time__1312372323000000 1312372323000000
changetime__1314539147000000 1314539147000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [32476]) Branching to 'required-ifaces-skip-msg-5217'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

Nothing very complicated here.  [Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/required-ifaces-skip-msg-5217).

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)<a name="note_3"></a>|@mwhudson commented|
|-|-|

Looks fine to me, please land.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [32480]) Merge required-ifaces-skip-msg-5217

Author: exarkun
Reviewer: mwh
Fixes: [#5217](https://github.com/twisted/twisted/issues/5217)

Improve the skip message for `ReactorBuilder`-based tests when the reactor is missing
a required interface for a particular test.
